osu.Game/OsuGame.cs

```diff
@@ -573,7 +573,9 @@ protected override void LoadComplete()
                             Origin = Anchor.BottomLeft,
                             Action = () =>
                             {
-                                if ((ScreenStack.CurrentScreen as IOsuScreen)?.AllowBackButton == true)
+                                var currentScreen = ScreenStack.CurrentScreen as IOsuScreen;
+
+                                if (currentScreen?.AllowBackButton == true && !currentScreen.OnBackButton())
```

Comment:
I'm not sure I'm following why both `AllowBackButton` and `OnBackButton` need to exist. `OnBackButton` seems like the more general mechanism and can probably supersede all usages of `AllowBackButton`, unless I'm missing something.

Reply:
`AllowBackButton` is there to hide the back button and disable its action on certain screens (like player doesn't need a back button, and has local exit logic).

`OnBackButton` is there to override the action, and only logical to implement if `AllowBackButton` is true.